modules/http4s/src/smithy4s/http4s/SimpleRestJsonBuilder.scala

```diff
@@ -17,28 +17,47 @@
 package smithy4s
 package http4s
 
-object SimpleRestJsonBuilder extends SimpleRestJsonBuilder(1024, false)
+object SimpleRestJsonBuilder extends SimpleRestJsonBuilder(1024, false, true)
 
 class SimpleRestJsonBuilder private (
     simpleRestJsonCodecs: internals.SimpleRestJsonCodecs
 ) extends SimpleProtocolBuilder[alloy.SimpleRestJson](
       simpleRestJsonCodecs
     ) {
 
-  def this(maxArity: Int, explicitDefaultsEncoding: Boolean) =
-    this(new internals.SimpleRestJsonCodecs(maxArity, explicitDefaultsEncoding))
+  def this(
+      maxArity: Int,
+      explicitDefaultsEncoding: Boolean,
+      hostPrefixInjection: Boolean
+  ) =
+    this(
+      new internals.SimpleRestJsonCodecs(
+        maxArity,
+        explicitDefaultsEncoding,
+        hostPrefixInjection
+      )
+    )
 
   def withMaxArity(maxArity: Int): SimpleRestJsonBuilder =
     new SimpleRestJsonBuilder(
       maxArity,
-      simpleRestJsonCodecs.explicitDefaultsEncoding
+      simpleRestJsonCodecs.explicitDefaultsEncoding,
+      simpleRestJsonCodecs.hostPrefixInjection
     )
 
   def withExplicitDefaultsEncoding(
       explicitDefaultsEncoding: Boolean
   ): SimpleRestJsonBuilder =
     new SimpleRestJsonBuilder(
       simpleRestJsonCodecs.maxArity,
-      explicitDefaultsEncoding
+      explicitDefaultsEncoding,
+      simpleRestJsonCodecs.hostPrefixInjection
+    )
+
+  def disableHostPrefixInjection(): SimpleRestJsonBuilder =
```

Comment:
I don't think this customisation is needed at this level. What's your rationale for exposing it ?

Reply:
I also thought that, but when looking up something in the spec, I saw: 

> Clients SHOULD provide a way for users to disable the hostPrefix injection behavior.

Reply:
Yeah I just copied it over from the other PR/branch, but I kept it as I saw the language David quoted there